lib/collision/index.d.ts

```diff
@@ -45,13 +45,52 @@ export interface AABB {
 }
 
 export let AABB: {
-  new(lower: Vec2, upper: Vec2): AABB;
-     (lower: Vec2, upper: Vec2): AABB;
+  new(lower?: Vec2, upper?: Vec2): AABB;
+     (lower?: Vec2, upper?: Vec2): AABB;
 
   isValid(o: any): boolean;
   assert(o: any): void;
-  extend(aabb: AABB, value: number): void;
+  extend(aabb: AABB, value: number): AABB;
   testOverlap(a: AABB, b: AABB): boolean;
   areEqual(a: AABB, b: AABB): boolean;
   diff(a: AABB, b: AABB): number;
 };
+
+export interface TreeNode {
+  id: string;
+  aabb: AABB;
+  userData: unknown;
+  parent: TreeNode;
+  child1: TreeNode;
+  child2: TreeNode;
+  height: number;
+}
+
+export interface DynamicTree {
+}
+
+export let DynamicTree: {
+  new(): DynamicTree;
+
+  getUserData(id: string): unknown;
+  getFatAABB(id: string): AABB;
+  allocateNode(): TreeNode;
+  freeNode(node: TreeNode): void;
+  createProxy(aabb: AABB, userData: unknown): string;
+  destroyProxy(id: string): void;
+  moveProxy(id: string, aabb: AABB, d: Vec2): boolean;
+  insertLeaf(leaf: TreeNode): void;
+  removeLeaf(leaf: TreeNode): void;
+  balance(iA: TreeNode): TreeNode;
+  getHeight(): number;
+  getAreaRatio(): number;
+  computeHeight(id: string): number;
+  validateStructure(node: TreeNode): void;
+  validateMetrics(node: TreeNode): void;
+  validate(): void;
+  getMaxBalance(): number;
+  rebuildBottomUp(): void;
+  shiftOrigin(newOrigin: Vec2): void;
+  query(aabb: AABB, queryCallback: (id: string) => boolean): void;
+  rayCast(input: RayCastInput, rayCastCallback: (subInput: RayCastInput, id: string) => number): void;
```

Comment:
All these methods should be in the Interface, because they are defined on the prototype of `DynamicTree`.